### PR TITLE
travis: cache build results with ccache

### PR DESCRIPTION
**Pull request by authmillenon**
Inspiration:
 - [The travis documentation](http://docs.travis-ci.com/user/caching/)
 - [The ccache manual](https://ccache.samba.org/manual.html)

### CODE REVIEW

**Comment by phiros** [comment]
I don't think that adding something to the travis cache will do anything for us. The travis documentation says:

Caching Dependencies and Directories
"The features described here are currently **only available for private repositories on travis-ci.com and our new container-based infrastructure**" 

AFAIK we use neither. 

BTW: I already did some experiments with ccache in the past (don't ask me in which PR) and found
that it actually increased the build time quite significantly for the first build. That combined with the fact
that we cannot use the cache feature of travis means that we would end up building a compiler cache (at a heavy performance price) every time without using it ever again...

File: .travis.yml [hunk 1]
```diff
@@ -33,6 +35,20 @@ install:
...
+    - ln -s /usr/bin/ccache /usr/local/bin/c++
+    - ln -s /usr/bin/ccache /usr/local/bin/arm-none-eabi-gcc
+    - ln -s /usr/bin/ccache /usr/local/bin/arm-none-eabi-g++
+    - ln -s /usr/bin/ccache /usr/local/bin/arm-none-eabi-c++
+    - ln -s /usr/bin/ccache /usr/local/bin/avr-gcc
+    - ln -s /usr/bin/ccache /usr/local/bin/avr-g++
+    - ln -s /usr/bin/ccache /usr/local/bin/avr-c++
+    - ln -s /usr/bin/ccache /usr/local/bin/msp430-gcc
+    - ln -s /usr/bin/ccache /usr/local/bin/msp430-g++
+    - ln -s /usr/bin/ccache /usr/local/bin/msp430-c++
```
Comment (phiros) [inline .travis.yml]: The ccache documentation is a little bit misleading (for debian/ubuntu users). AFAIK you don't have to create those symlinks by hand. On a debian/ubuntu system there is a directory (```/usr/lib/ccache```) already populated with symlinks for all installed compilers on your system. You just have prepend that directory to your PATH.

**Comment by authmillenon** [comment]
So maybe its best to remove the `cache: apt` line too?

**Comment by phiros** [comment]
yeah probably. I think this is still an artifact of an experiment @Kijewski did.

**Comment by phiros** [comment]
Since this isn't fixable it probably doesn't make sense to leave this PR open.
Any objections to that?

**Pull request closed without merging**